File: app/src/main/java/ru/deltadelete/lab10/entities/TownWithDistricts.java
package ru.deltadelete.lab10.entities;

import androidx.room.Embedded;
import androidx.room.Relation;

import java.util.List;

public class TownWithDistricts {
    @Embedded
    public Town town;
    @Relation(parentColumn = "town_id", entityColumn = "town_id")
    public List<District> districts;
}

File: app/src/main/java/ru/deltadelete/lab11/database/entities/TownWithDistricts.kt
package ru.deltadelete.lab11.database.entities

import androidx.room.Embedded
import androidx.room.Relation

public data class TownWithDistricts(

    @Embedded
    var town: Town? = null,

    @Relation(parentColumn = "town_id", entityColumn = "town_id")
    var districts: List<District>? = null,
)
